src/webview/columns/data-column.tsx

```diff
@@ -31,67 +55,240 @@ export class DataColumn implements ColumnContribution {
-        return groups;
-    }
 
-    protected renderWord(memory: Memory, options: TableRenderOptions, currentAddress: bigint): React.ReactNode {
-        const initialOffset = toOffset(memory.address, currentAddress, options.bytesPerWord * 8);
-        const finalOffset = initialOffset + options.bytesPerWord;
-        const bytes: React.ReactNode[] = [];
-        for (let i = initialOffset; i < finalOffset; i++) {
-            bytes.push(this.renderEightBits(memory, currentAddress, i));
-        }
-        this.applyEndianness(bytes, options);
-        return <span className='single-word' key={currentAddress.toString(16)}>{bytes}</span>;
+        if (words.length) { byteGroups.push({ startAddress, endAddress: range.endAddress, words }); }
+
+        return byteGroups;
     }
 
     protected applyEndianness<T>(group: T[], options: TableRenderOptions): T[] {
         // Assume data from the DAP comes in Big Endian so we need to revert the order if we use Little Endian
         return options.endianness === Endianness.Big ? group : group.reverse();
     }
+}
+
+export interface EditableDataColumnGroupProps {
+    range: BigIntMemoryRange;
+    group: DataColumnByteGroup;
+    options: TableRenderOptions;
+    writeMemory?(writeArguments: DebugProtocol.WriteMemoryArguments): Promise<void>;
+}
+
+export interface EditableDataColumnGroupState {
+    value: string;
+    isEdit: boolean;
+}
+
+export class EditableDataColumnGroup extends React.Component<EditableDataColumnGroupProps, EditableDataColumnGroupState> {
+
+    constructor(props: EditableDataColumnGroupProps) {
+        super(props);
+
+        this.state = {
+            isEdit: false,
+            value: this.asLine(this.props.group)
+        };
+    }
+
+    protected byteGroupStyle: React.CSSProperties = {
+        marginRight: `${DataColumn.Styles.MARGIN_RIGHT_PX}px`
+    };
+
+    protected get renderableCharacters(): number {
+        return this.props.options.bytesPerWord * this.props.options.wordsPerGroup * 2;
+    }
 
-    protected renderEightBits(memory: Memory, currentAddress: bigint, offset: number): React.ReactNode {
-        const { content, className, style, title } = this.getBitAttributes(memory, currentAddress, offset);
+    render(): React.ReactNode {
+        if (this.state.isEdit) {
+            return this.renderEditingGroup();
+        }
+
+        return this.renderReadonlyGroup();
+    }
+
+    componentDidUpdate(_prevProps: Readonly<EditableDataColumnGroupProps>, prevState: Readonly<EditableDataColumnGroupState>): void {
+        if (prevState.isEdit === false && this.state.isEdit) {
+            this.setState(prev => ({ ...prev, value: this.asLine(this.props.group) }));
+        }
+    }
+
+    protected asLine(group: DataColumnByteGroup): string {
+        return group.words.flatMap(w => w.bits.flatMap(b => b.value).join('')).join('');
```

Comment:
I believe the inner `.join()` is likely unnecessary, as `flatMap` handles the various levels of array wrapping.